apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2278,17 +2410,51 @@ static void libc_stdio_rawoutstream_tc(void)
 static void libc_stdio_rawsistream_tc(void)
 {
 	int fd;
-	char *filename = VFS_FILE_PATH;
+	int ret;
+	int getch;
+	off_t offset;
+	char *buf = STREAM_TEST_CONTENTS;
 
 	struct lib_rawsistream_s rawsistream;
 
-	fd = open(filename, O_RDONLY);
+	/* Negative case, no character is read from the rawsistream */
+
+	fd = open(VFS_FILE_PATH, O_WRONLY);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", rawsistream.fd, fd, close(fd));
+
+	getch = rawsistream.public.get((FAR struct lib_sistream_s *)&rawsistream.public);
+	close(fd);
+	TC_ASSERT_EQ("rawsistream_getc", rawsistream.public.nget, 0);
+	TC_ASSERT_EQ("rawsistream_getc", getch, EOF);
+
+	/* Positive case, one character is read from the rawsistream */
+
+	fd = open(VFS_FILE_PATH, O_WRONLY | O_TRUNC);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	ret = write(fd, buf, strlen(buf));
+	TC_ASSERT_EQ_CLEANUP("write", ret, strlen(buf), close(fd));
+	close(fd);
+
+	fd = open(VFS_FILE_PATH, O_RDONLY);
 	TC_ASSERT_GEQ("open", fd, 0);
 
 	lib_rawsistream((FAR struct lib_rawsistream_s *)&rawsistream, fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", rawsistream.fd, fd, close(fd));
+
+	getch = rawsistream.public.get((FAR struct lib_sistream_s *)&rawsistream.public);
+	TC_ASSERT_EQ_CLEANUP("rawsistream_getc", rawsistream.public.nget, 1, close(fd));
+	TC_ASSERT_EQ_CLEANUP("rawsistream_getc", getch, 'T', close(fd));
+
+	/* Check seek operation */
+
+	offset = rawsistream.public.seek((FAR struct lib_sistream_s *)&rawsistream.public, SEEK_OFFSET, SEEK_DEF);
+	TC_ASSERT_EQ_CLEANUP("lib_rawsistream", rawsistream.fd, fd, close(fd));
 	close(fd);
-	TC_ASSERT_EQ("lib_rawsistream", rawsistream.fd, fd);
-	TC_ASSERT_EQ("lib_rawsistream", rawsistream.public.nget, 0);
+	TC_ASSERT_EQ("rawsistream_seek", offset, (off_t)SEEK_OFFSET);
```

Comment:
SEEK_DEF is invalid parameter in seek function.
How can both variables be same?